deep_qa/data/instances/sequence_tagging/verb_semantics_instance.py

```diff
@@ -0,0 +1,208 @@
+        state_change: str
+            state change type
+        arg1: (int, int)
+            span of the argument1 of the state change
+        arg2: (int, int)
+            Span of the argument2 of the state change
+        E.g. For the same example above,
+            sentence = ["roots", "absorb", "water", "from", "the", "soil"]
+            verb = (1,1)   // absorb
+            entity = (2,2) // water
+            arg1 = (4,5)   // the soil
+            arg2 = (0,0)   // roots
+        """
+        self.sentence = sentence
+        self.verb = verb
+        self.entity = entity
+        tags = self.__make_tag_sequence(arg1, arg2)
+        super(VerbSemanticsInstance, self).__init__((state_change, tags), index)
+
+
+    def __str__(self):
+        return 'VerbSemanticsInstance(' + self.sentence + ', ' + self.verb + ', ' + \
+               self.entity + str(self.label) + ')'
+
+    def __make_tag_sequence(self, arg1: (int, int), arg2: (int, int)) -> List[str]:
+        """
+        Convert (arg1, arg2) into a tag sequence over the sentence
+        arg1: (int, int)
+            StartIndex,endIndex of arg1 in the sentence
+        arg2: (int, int)
+            StartIndex,endIndex of arg2 in the sentence
+        tag sequence: List[str]
+            List of tag names one per token in the sentence
+        E.g. sentence = "Roots absorb water from the soil"
+            arg1 = (4,5)  // the soil
+            arg2 = (0,0)  // roots
+            tag sequence = (B-ARG2, O, O, O, B-ARG1, I-ARG1)
+        """
+        tags = ["O"] * len(self.sentence)
+        if arg1[0] != -1:
+            tags[arg1[0]] = "B-" + "ARG1"
+            for index1 in range(arg1[0]+1, arg1[1]):
+                tags[index1] = "I-" + "ARG1"
+        if arg2[0] != -1:
+            tags[arg2[0]] = "B-" + "ARG2"
+            for index2 in range(arg2[0]+1, arg2[1]):
+                tags[index2] = "I-" + "ARG2"
+        return tags
+
+    @overrides
+    def words(self) -> Dict[str, List[str]]:
+        """
+        Creates 3 different word dictionaries:
+        1) words in the sentence
+        2) state change types
+        3) tags for arguments
+        """
+        words = self._words_from_text(self.sentence)
+        words['state_changes'] = [self.label[0]]
+        words['tags'] = self.label[1]
+        return words
+
+    @overrides
+    def to_indexed_instance(self, data_indexer: DataIndexer):
+
+        sentence_indices = [data_indexer.get_word_index(word) for word in self.sentence]
+
+        # verb indices, entity indices are one-hot vectors representing spans within the sentence
+        verb_indices = numpy.zeros(len(sentence_indices))
+        for entity_id in range(self.verb[0], self.verb[1] + 1):
+            verb_indices[entity_id] = 1
+
+        entity_indices = numpy.zeros(len(sentence_indices))
+        for entity_id in range(self.entity[0], self.entity[1] + 1):
+            entity_indices[entity_id] = 1
+
+        # indexed_label = (state_change, tags)
```

Comment:
This looks too much like commented out code - can you make it more obvious it's a comment, by adding some English words to it?